### DISCUSSION TITLE
課題14についての質問（8/16)

### DISCUSSION THREAD

**Discussion by TakedaAyano** [nsc-kiwamu/traning-takeda#70] [Q&A]
以下のjudgeSosuu関数を2つの問題で使い回すということは、「 // 素数判定の処理を追加する」の部分に、素数判定の処理を記述することによって、問題１と２で記述するコードは短くなるということでしょうか？

/**
  * 素数の判断をする
  * 素数の場合は引数で受け取った値を返却
  * 素数でない場合はundedefindを返却
  * @param target 素数判定したい値
  * @return 素数判定結果
  */
 function judgeSosuu(target: number):number | undefined {
  let retNum:number | undefined

  // 素数判定の処理を追加する

  return retNum
}

**Comment by nsc-kiwamu** [comment]
そうですね、目的は「やりたいことの分離」をしてコードを短くしたいです。

例えば問題1では
「2～100のうち素数を見つける」という処理を実現したいのですが、そのために
(1)2～100まで繰り返す
(2)渡された値が素数かどうか判断する
と処理を分離したくて、(2)が関数化したい箇所です。

それができれば、問題2では
「要素数10個のランダムな値の中から素数を見つける」という処理を実現するために
(1)要素数10の配列分処理を繰り返す
(2)渡された値が素数かどうか判断する
と処理を分離すれば、(2)の関数化したところは問題1でも2でも使えると思います。

ちなみに問題1ですが、以下の処理になっています。
```
    for(let n=2; n < num; n++){ // nが2以上の時num(100)まで繰り返す。
      let ans = judgeSosuu(0)

      if (typeof ans === 'number') {
        console.log(`${ans}は素数です`)
      }
    }
```
これだと「2～100まで繰り返し、0が素数か判断する」という処理になるので
NGだということは理解できますか？(0が素数かどうかを2～100まで(=99回)繰り返し判定しているだけ)
その前は「judgeSosuu(num)」だったので
「2～100まで繰り返し、100が素数か判断する」
となるので、これもNGですよね？

また、作りかけだとは思いますが、関数部分
```
 function judgeSosuu(target: number):number | undefined {
  let retNum:number | undefined
  let n,i;
  // 素数判定の処理を追加する
  for(let i=2; i<n; i++){
        if(n%i == 0){ // n÷iの余りがなければ、
              break; // 終了
        }
    if(n == i) {// nとiが同じであれば、素数
    }

  return retNum
 }
}
```
引数で受け取った値が素数かどうか判断する。
となるのですが、引数で受け取った値を使っていないですよね？
「引数で受け取った値が素数かどうか」
を判断してください。
